platform/broadcom/sonic-platform-modules-ragile/common/script/fancontrol.py

```diff
@@ -0,0 +1,991 @@
+#!/usr/bin/env python3
```

Comment:
Why a bespoke fan control program? Why not utilize thermalctld?

Reply:
The reason is that this file conclude our own defined speed control strategy:
1. We want anti-shake mechanism
2. There is a monitoring MAC temperature, there is a restart mechanism
3. Others

Reply:
Understood. However, these are enhancements that would be nice-to-haves in SONiC as a whole in the future, and vendors can implement them if they support them. Please feel free to create a HLD and open a PR in https://github.com/Azure/SONiC/pulls to propose adding features to thermalctld or any other SONiC application.

Reply:
OK, glad to hear that. We will plan to submit a HLD and make contribution to making these features common to SONiC next stage.